portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -53,21 +55,21 @@ private void writeToDatabase(Entity toWrite) {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String limitString = getParameter(request, "number-comments", "5");
 
-    List<Comment> comments = getComments();
+    List<Comment> comments = getComments(Integer.parseInt(limitString));
 
```

Comment:
Adding to the previous discussion on parsing, though indeed, through your interface's users can't select negative or non-numeric values, they can manually call these methods here in the servlet backend by mimicking your post requests (but with bad values).

I'd suggest catching the NumberFormatException that parseInt can generate, and also checking it's of a sane size. Not checking for this can allow attackers to bring down your service (for instance, by requesting you serve millions of comments per request, and thousands of requests per second).

Reply:
That makes sense. I updated it so that it catches `NumberFormatException` and also makes sure the size is between 1 and 100. Thank you!